lib/routes/emails.js

```diff
@@ -225,8 +225,25 @@ module.exports = (log, db, mailer, config, customs, push) => {
                   return reply({})
                 }
               } else if (sessionToken.tokenVerificationId) {
+
                 emails = emailData
                 code = sessionToken.tokenVerificationId
+
+                // Check to see if this account has a verified TOTP token. If so, then it should
+                // not be allowed to bypass TOTP requirement by sending a sign-in confirmation email.
+                return db.totpToken(sessionToken.uid)
+                  .then((result) => {
+                    if (result && result.verified && result.enabled) {
+                      throw error.unverifiedSession()
```

Comment:
IMO it feels a bit weird to throw an error here; what do you think about copying the behaivour from higher up in the function [1] and just silently succeeding without sending the email?

[1] https://github.com/mozilla/fxa-auth-server/pull/2365/files#diff-cf65808c0d0da40a5f824db418c4ccd7R157